### PR TITLE
[ticket/16698] Enhance check for default charset

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-16698

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

https://tracker.phpbb.com/browse/PHPBB3-16698

### CODE REVIEW

**Comment by EA117** [comment]
Not something that "must" be considered in this change.  But is this a good opportunity to also account for the fact that having default_charset "not set at all" (meaning it doesn't exist in the PHP.INI file at all) also means "PHP APIs will default to UTF-8"?

That wasn't always true, but is true in the versions of PHP that phpBB 3.3.0 and later support: https://www.php.net/manual/en/ini.core.php#ini.default-charset

The reason to mention it now is because it causes the same situation for phpBB that ticket 16698 is in the middle of fixing: Nothing is actually "wrong" because the PHP APIs are defaulting to UTF-8 when the PHP.INI setting doesn't exist; but phpBB's test of "what is the PHP.INI setting set to" is still failing, because we don't find that it's set to the value "UTF-8".

It also seems worth at least consider, since this default_charset test is new to phpBB 3.3.3, and this can be the right time to make sure these edge cases aren't triggering false positives of the new message.

**Comment by 3D-I** [comment]
Setting default_charset to an empty value is not recommended.

**Comment by EA117** [comment]
Agreed.  But that's not the same thing as "does not exist in the PHP.INI file at all."

**Comment by rxu** [comment]
Tend to agree with @EA117 as we have support requests about the warning from users who has no `default_charset` parameter in `php.ini` but their boards operate well.

File: phpBB/includes/acp/acp_main.php [hunk 1]
```diff
@@ -704,7 +704,7 @@ function main($id, $mode)
 				'S_MBSTRING_ENCODING_TRANSLATION_FAIL'	=> $encoding_translation && ($encoding_translation != 0),
 				'S_MBSTRING_HTTP_INPUT_FAIL'			=> !empty($http_input),
 				'S_MBSTRING_HTTP_OUTPUT_FAIL'			=> !empty($http_output),
-				'S_DEFAULT_CHARSET_FAIL'				=> $default_charset !== 'UTF-8',
+				'S_DEFAULT_CHARSET_FAIL'				=> $default_charset !== 'utf-8',
```
Comment (rxu) [inline phpBB/includes/acp/acp_main.php:707]:
```suggestion
				'S_DEFAULT_CHARSET_FAIL'				=> $default_charset !== null && $default_charset !== 'utf-8',
```

Comment (3D-I) [reply phpBB/includes/acp/acp_main.php]: Did it differently since `$this->php_ini->getString()` can return NULL but `strtolower()` returns a NULL value as an empty string. As per https://www.php.net/manual/en/function.strtolower.php#116230

**Review by CHItA — APPROVED** [review]
